libs/utils/src/tabulation/lookups.ts

```diff
@@ -149,12 +149,10 @@ export function determinePartyId<T>(
   group: Tabulation.GroupOf<T>
 ): Optional<string> {
   if (group.partyId) return group.partyId;
-
   if (!group.ballotStyleGroupId) return undefined;
-  const ballotStyle = getGroupedBallotStyles(
-    electionDefinition.election.ballotStyles
-  ).find((bs) => bs.id === group.ballotStyleGroupId);
-  if (!ballotStyle) return undefined;
-
-  return ballotStyle.partyId;
+  const ballotStyleGroup = getBallotStyleGroup({
+    election: electionDefinition.election,
+    ballotStyleGroupId: group.ballotStyleGroupId as BallotStyleGroupId,
+  });
+  return ballotStyleGroup?.partyId;
```

Comment:
I thought it was against the lint rules to use the non-null ? assertion. Did I just totally make that up?

Reply:
We can't use `!` but we can use `?`